### PR TITLE
Mosviz: Slit overlay to use Data WCS

### LINKED ISSUE #849: [BUG] Mosviz: loading images throws error
Loading an image in via `load_data` in Mosviz after merge of https://github.com/spacetelescope/jdaviz/pull/783, throws an error.  The following code snippet loads the spectra and image file, albeit with some warnings printed.   (It's noted that the below code does not display the image data by default.  I have to manually check the image in the data dropdown menu)
```
m = MosViz()
m.app
m.load_data(spectra_1d=n1d, spectra_2d=n2d, images=None)
m.load_images(images[1])
```
but the following code throws the traceback error below.  The image data still loads, and is automatically selected for display in the image viewer, but the image scaling is different than in the above case.
```
m = MosViz()
m.app
m.load_data(spectra_1d=n1d, spectra_2d=n2d, images=[images[1]])
```

n1d and n2d are each a list of spectra 1d and 2d files and images[1] is a single image file, JWST `i2d` FITS.  Tried with both a MIRI and NIRISS `i2d` file. 

Full traceback
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-22-229cbb2b2b93> in <module>
----> 1 m.load_data(spectra_1d=n1d, spectra_2d=n2d, images=images[0])

~/Work/git/havok2063/jdaviz/jdaviz/configs/mosviz/helper.py in load_data(self, spectra_1d, spectra_2d, images, spectra_1d_label, spectra_2d_label, images_label, *args, **kwargs)
    268 
    269         # Load the first object into the viewers automatically
--> 270         self.app.get_viewer("table-viewer").figure_widget.highlighted = 0
    271 
    272     def link_table_data(self, data_obj):

~/anaconda3/envs/havokve/lib/python3.8/site-packages/traitlets/traitlets.py in __set__(self, obj, value)
    602             raise TraitError('The "%s" trait is read-only.' % self.name)
    603         else:
--> 604             self.set(obj, value)
    605 
    606     def _validate(self, obj, value):

~/anaconda3/envs/havokve/lib/python3.8/site-packages/traitlets/traitlets.py in set(self, obj, value)
    591             # we explicitly compare silent to True just in case the equality
    592             # comparison above returns something other than True/False
--> 593             obj._notify_trait(self.name, old_value, new_value)
    594 
    595     def __set__(self, obj, value):

~/anaconda3/envs/havokve/lib/python3.8/site-packages/traitlets/traitlets.py in _notify_trait(self, name, old_value, new_value)
   1215 
   1216     def _notify_trait(self, name, old_value, new_value):
-> 1217         self.notify_change(Bunch(
   1218             name=name,
   1219             old=old_value,

~/anaconda3/envs/havokve/lib/python3.8/site-packages/ipywidgets/widgets/widget.py in notify_change(self, change)
    604                 # Send new state to front-end
    605                 self.send_state(key=name)
--> 606         super(Widget, self).notify_change(change)
    607 
    608     def __repr__(self):

~/anaconda3/envs/havokve/lib/python3.8/site-packages/traitlets/traitlets.py in notify_change(self, change)
   1225     def notify_change(self, change):
   1226         """Notify observers of a change event"""
-> 1227         return self._notify_observers(change)
   1228 
   1229     def _notify_observers(self, event):

~/anaconda3/envs/havokve/lib/python3.8/site-packages/traitlets/traitlets.py in _notify_observers(self, event)
   1262                 c = getattr(self, c.name)
   1263 
-> 1264             c(event)
   1265 
   1266     def _add_notifiers(self, handler, name, type):

~/Work/git/havok2063/jdaviz/jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py in place_slit_overlay(self, *args, **kwargs)
     92 
     93                 # Use wcs of image viewer to scale slit dimensions correctly
---> 94                 wcs_image = WCS(image_data[0].meta)
     95                 pixel_region = sky_region.to_pixel(wcs_image)
     96 

~/anaconda3/envs/havokve/lib/python3.8/site-packages/astropy/wcs/wcs.py in __init__(self, header, fobj, key, minerr, relax, naxis, keysel, colsel, fix, translate_units, _do_set)
    414                     header = fits.Header()
    415                     for dict_key in orig_header.keys():
--> 416                         header[dict_key] = orig_header[dict_key]
    417                 except TypeError:
    418                     raise TypeError(

~/anaconda3/envs/havokve/lib/python3.8/site-packages/astropy/io/fits/header.py in __setitem__(self, key, value)
    205             # If we get an IndexError that should be raised; we don't allow
    206             # assignment to non-existing indices
--> 207             self._update((key, value, comment))
    208 
    209     def __delitem__(self, key):

~/anaconda3/envs/havokve/lib/python3.8/site-packages/astropy/io/fits/header.py in _update(self, card)
   1657                 self._modified = True
   1658         elif keyword in Card._commentary_keywords:
-> 1659             cards = self._splitcommentary(keyword, value)
   1660             if keyword in self._keyword_indices:
   1661                 # Append after the last keyword of the same type

~/anaconda3/envs/havokve/lib/python3.8/site-packages/astropy/io/fits/header.py in _splitcommentary(self, keyword, value)
   1899             cards = []
   1900             while idx < len(valuestr):
-> 1901                 cards.append(Card(keyword, valuestr[idx:idx + maxlen]))
   1902                 idx += maxlen
   1903         return cards

~/anaconda3/envs/havokve/lib/python3.8/site-packages/astropy/io/fits/card.py in __init__(self, keyword, value, comment, **kwargs)
    191                 self.keyword = keyword
    192             if value is not None:
--> 193                 self.value = value
    194 
    195         if comment is not None:

~/anaconda3/envs/havokve/lib/python3.8/site-packages/astropy/io/fits/card.py in value(self, value)
    328             m = self._ascii_text_re.match(value)
    329             if not m:
--> 330                 raise ValueError(
    331                     'FITS header values must contain standard printable ASCII '
    332                     'characters; {!r} contains characters not representable in '

ValueError: FITS header values must contain standard printable ASCII characters; '\nAssociation information\n\n\nProgram information\n\n\nObservation identifiers' contains characters not representable in ASCII or non-printable characters.
```

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to have slit overlay plugin to not try to recreate the WCS itself, but rather use the WCS or GWCS that already comes with the data object.

Fixes #849 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py [hunk 1]
```diff
@@ -78,11 +77,11 @@ def place_slit_overlay(self, *args, **kwargs):
         self.remove_slit_overlay()
 
         # Get data from relevant viewers
-        image_data = self.app.get_viewer("image-viewer").data()
+        image_data = self.app.get_viewer("image-viewer").state.reference_data
```
Comment (pllim) [inline jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:80]: I don't know why original code wanted `image_data[0]` specifically. I just assume `reference_data` is the same thing. If not, please let me know. Thanks!

File: setup.cfg [hunk 2]
```diff
@@ -53,7 +53,7 @@ docs =
     sphinx-astropy
 all =
     scikit-image
-    regions
+    regions>=0.5
```
Comment (pllim) [inline setup.cfg:56]: Need this minversion for `regions` to support GWCS.

File: jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py (hunk 1 above)
Comment (eteq) [reply jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:80]: I think this was an assumption about the data format, so I think you are doing the right thing here!

**Review by eteq — APPROVED** [review]
I gave this a  look and it looks right to me.  But @rosteen  or @javerbukh might want to have a look just to make sure there's not some hidden assumptions in the data that I'm missing.

**Review by ibusko — COMMENTED** [review]
What's a good data set to test this?

File: jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py (hunk 1 above)
Comment (rosteen) [reply jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:80]: Yeah, this works.

**Review by rosteen — APPROVED** [review]
Looks good, thanks.

**Pull request merged**